doc/development/guide/logging.rst

```diff
@@ -0,0 +1,407 @@
+==============================
+
+Enabling logging
+----------------
+
+To enable logging support in your PennyLane work-flow simply run the following lines:
+
+.. code:: python
+
+   import pennylane as qml
+   qml.logging.enable_logging()
+
+
+This will ensure all levels of the execution pipeline logs function entries, and
+outputs to handlers pointing to the standard output.
+
+Adding logging supports during development
+------------------------------------------
+
+To add logging support to components of PennyLane, we must define a module logger and ensure all scoped log statements use that logger. A logger can be imported and defined with:
+
+.. code:: python
+
+   import logging
+   logger = logging.getLogger(__name__)
+
+which will be used within the given module, and track directories,
+filenames and function names, as we have defined the appropriate types
+within the formatter configuration (see
+``pennylane.logging.formatters.formatter.py::DefaultFormatter``). With
+the logger defined, we can selectively add to the logger by if-else
+statements, which compare the given module’s log-level to any log record
+message it receives. This step is not necessary, as the message will
+only output if the level is enabled, though if an expensive function
+call is required to build the string for the log-message, it can be
+faster to perform this check:
+
+.. code:: python
+
+   if logger.isEnabledFor(logging.DEBUG):
+       logger.debug(
+           """Entry with args=(arg_name_1=%s, arg_name_2=%s, ..., arg_name_n=%s)""",
+           arg_name_1, arg_name_2, ..., arg_name_n,
+       )
+
+The above line can be added to below the function/method entry point,
+and the provided arguments can be used to populate the log message. This
+allows us a way to track the inputs and calls through the stack in the
+order they are executed, as is the basis for following a trail of
+execution as needed.
+
+All debug log-statements currently added to the PennyLane execution
+pipeline output log level messages with the originating function making
+that call, as this can allow us to track hard-to-find bug origins in the
+execution pipeline.
+
+
+Customizing the logging configuration
+-------------------------------------
+
+As with any package that targets many domains, Python’s logging is as
+extensible and flexible as it is a challenge to configure for all user needs 
+– ideally we define some good defaults that meet our development goals, 
+and only deviate from them if required. For further details and customization 
+options I will suggest reading the Python logging documentation
+`how-to <https://docs.python.org/3/howto/logging.html#python%20logging>`__
+and `“advanced”
+tutorial <https://docs.python.org/3/howto/logging.html#logging-advanced-tutorial>`__
+level. 
+
+PennyLane logging defaults are contained in a configuration file in the logging module titled 
+``log_config.toml``, which is imported when logging is enabled. To change log-levels that are 
```

Comment:
Where can a user find this file? Shall we mention `config_path`?

Reply:
Added the path info call.